### PR TITLE
VB: Remove unnecessary check for ModuleBuilderOpt during lowering phase

### PR DESCRIPTION
**Pull request by wschae**
These checks were required for some VB tests, not for product code.
By updating VB tests, we can remove them and simplify the lowering phase.

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.vb [hunk 1]
```diff
@@ -85,9 +85,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             stateMachineType = New AsyncStateMachine(slotAllocatorOpt, compilationState, method, methodOrdinal, kind)
 
-            If compilationState.ModuleBuilderOpt IsNot Nothing Then
-                compilationState.ModuleBuilderOpt.CompilationState.SetStateMachineType(method, stateMachineType)
-            End If
+            Debug.Assert(compilationState.ModuleBuilderOpt IsNot Nothing)
+            compilationState.ModuleBuilderOpt.CompilationState.SetStateMachineType(method, stateMachineType)
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.vb]: So much nicer

**Comment by jaredpar** [comment]
LGTM

**Comment by wschae** [comment]
@tmat @AlekseyTs Can you take a look?

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb [hunk 2]
```diff
@@ -11741,16 +11739,15 @@ End Class
...
-
-            AssertTheseDiagnostics(compilation,
-<expected>
-BC40054: 'Public Sub New(a As Integer)' in designer-generated type 'FromDesigner1' should call InitializeComponent method.
-    Sub New(a As Integer)
-        ~~~
-BC40054: 'Public Sub New(c As Integer)' in designer-generated type 'FromDesigner5' should call InitializeComponent method.
-    Sub New(c As Integer)
-        ~~~
-</expected>)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb]: Please use a helper that allows to keep base-line in the original form.

Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb]: Applies to other similar changes.

File: src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.vb [hunk 3]
```diff
@@ -85,9 +85,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             stateMachineType = New AsyncStateMachine(slotAllocatorOpt, compilationState, method, methodOrdinal, kind)
 
-            If compilationState.ModuleBuilderOpt IsNot Nothing Then
-                compilationState.ModuleBuilderOpt.CompilationState.SetStateMachineType(method, stateMachineType)
-            End If
+            Debug.Assert(compilationState.ModuleBuilderOpt IsNot Nothing)
```
Comment (tmat) [inline src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.vb]: I think we can drop these assertion. ModuleBuilderOpt is now non-null during the entire lowering phase. Rather then assert so on every use add a doc comment on it saying so.

File: src/Compilers/VisualBasic/Portable/Lowering/LambdaRewriter/LambdaRewriter.vb [hunk 4]
```diff
@@ -307,27 +303,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
                     End If
 
-                    If CompilationState.ModuleBuilderOpt IsNot Nothing Then
-                        Dim frame = Me.lazyStaticLambdaFrame
+                    Dim frame = Me.lazyStaticLambdaFrame
 
-                        ' add frame type
-                        CompilationState.ModuleBuilderOpt.AddSynthesizedDefinition(_topLevelMethod.ContainingType, frame)
+                    ' add frame type
+                    CompilationState.ModuleBuilderOpt.AddSynthesizedDefinition(_topLevelMethod.ContainingType, frame)
```
Comment (tmat) [inline src/Compilers/VisualBasic/Portable/Lowering/LambdaRewriter/LambdaRewriter.vb]: Is this just github or is the indentation really off?

**Comment by tmat** [comment]
:+1:

File: src/Compilers/VisualBasic/Portable/Lowering/LambdaRewriter/LambdaRewriter.vb (hunk 4 above)
Comment (wschae) [inline src/Compilers/VisualBasic/Portable/Lowering/LambdaRewriter/LambdaRewriter.vb]: Indentation has been changed after removing if-clause :)

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb (hunk 2 above)
Comment (wschae) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb]: Sure. I will use AssertTheseEmitDiagnostics to keep those original forms. Thanks!

File: src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.vb (hunk 3 above)
Comment (wschae) [inline src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.vb]: Sure! Thanks!

**Comment by wschae** [comment]
@AlekseyTs Can you have a look at this update? I use AssertTheseEmitDiagnostics instead of VerifyEmitDiagnostics to keep the original baseline forms.

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb [hunk 5]
```diff
@@ -12710,15 +12708,15 @@ Class C
 End Class
 ]]></file>
 </compilation>)
-            comp.AssertTheseDiagnostics(
-<errors>
+            AssertTheseEmitDiagnostics(compilation,
+<errror>
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb]: Typo?

**Comment by AlekseyTs** [comment]
LGTM

File: src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb (hunk 5 above)
Comment (wschae) [inline src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTests.vb]: Good catch! Thanks!

**Comment by wschae** [comment]
Thanks!

**Pull request merged**